### DISCUSSION TITLE
toggle quit_on_open functionality with keymap

### DISCUSSION THREAD

**Discussion by manningc33** [nvim-tree/nvim-tree.lua#2757] [Q&A]
Is there a way to make a keymap that toggles nvim-trees quit on open functionality on and off? i.e. toggle nvim-tree.actions.open_file.quit_on_open. 

I'm pretty new to neovim, but I tried following these instructions, but didn't seem to work: https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes#toggle-adaptive-width not sure if it's because reload only reloads view settings and not action settings. This is my code: 

```lua
local quit_on_open = true

local toggle_quit_on_open = function()
  quit_on_open = not quit_on_open 
  require("nvim-tree.api").tree.reload()
end

vim.keymap.set("n", "<C-o>", toggle_quit_on_open, { noremap = true, desc = "toggle nvim-tree auto-close behavior" })

local get_quit_on_open = function()
  return quit_on_open
end

require("nvim-tree").setup({
  actions = {
    open_file = {
      quit_on_open = get_quit_on_open(),
    },
  },
})

```
I also saw on the example that in setup opts, they don't include the () for get_quit_on_open. When I do that, get an error saying that actions.open_file.quit_on_open. Expected boolean, got function

**Comment by alex-courtis** [comment]
> I'm pretty new to neovim, but I tried following these instructions, but didn't seem to work: https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes#toggle-adaptive-width not sure if it's because reload only reloads view settings and not action settings. This is my code:

That recipe specifically doesn't need to reload nvim-tree, which is very undesirable for performance, state etc. reasons.

There's currently no mechanism to dynamically change quit on open behaviour or any functionality like that. Adding such mechanisms is not trivial.

Options:
1. Create a mapping that calls `node.open.edit()` then optionally calls `tree.close()` depending on `local quit_on_open`
2. Add an option to `node.open.edit()`
```help
      • {close}          (boolean) close the tree after opening, default false
```
3. Add a runtime mechanism to change `actions.open_file.quit_on_open`

It would be most gratefully appreciated if you created a [PR](https://github.com/nvim-tree/nvim-tree.lua/blob/master/CONTRIBUTING.md) for 2 - it's the cleanest and most accessible solution.
